src/actor/component.rs

```diff
@@ -52,9 +55,18 @@ impl host::Host for Ctx<'_> {
         operation: String,
         payload: Option<Vec<u8>>,
     ) -> anyhow::Result<Result<Option<Vec<u8>>, String>> {
-        let invocation = (namespace, operation, payload)
+        let invocation: Invocation = (namespace, operation, payload)
             .try_into()
             .context("failed to parse invocation")?;
+        //TODO: ugly hacky try make better, maybe better .into above
```

Comment:
Agree with the comment here. I think you should be able to modify the `From<(a,b,c)>` implementation to add `From<(a,b,c,d)>` and then add the optional context to the last position (should be `Some(x)` or `None`).

Reply:
Yep! Doing that right now

Reply:
much cleaner now